src/mame/atari/atari400.cpp

```diff
@@ -1677,29 +1807,6 @@ LIGHT-ORANGE
  *
  **************************************************************/
 
-void a400_state::setup_ram(int bank, uint32_t size)
-{
-	offs_t ram_top;
-
-	switch (bank)
-	{
-	case 0: // 0x0000-0x7fff
-		ram_top = std::min(size, uint32_t(0x8000)) - 1;
-		m_maincpu->space(AS_PROGRAM).install_ram(0x0000, ram_top, m_ram->pointer());
-		break;
-	case 1: // 0x8000-0x9fff
-		ram_top = std::min(size, uint32_t(0xa000)) - 1;
-		if (ram_top > 0x8000)
-			m_maincpu->space(AS_PROGRAM).install_ram(0x8000, ram_top, m_ram->pointer() + 0x8000);
-		break;
-	case 2: // 0xa000-0xbfff
-		ram_top = std::min(size, uint32_t(0xc000)) - 1;
-		if (ram_top > 0xa000)
-			m_maincpu->space(AS_PROGRAM).install_ram(0xa000, ram_top, m_ram->pointer() + 0xa000);
-		break;
-	}
-}
```

Comment:
Installing RAM directly or using memory banks is better than using trampolines like you’ve added.  It’s substantially faster, and shows intent better when dumping memory maps in the debugger.  Are you sure you really need the trampolines?

Reply:
I definitely need to: RAM can be disabled by views and eventually by various overlays such as cart.

Reply:
That would just mean you need to add the ram in a view

Reply:
mtu-130 has a bunch of cases where the ram can be partially overriden by a rom, or by i/o, or put readonly in places.  I handle that with a pile of views.

Reply:
In this case for RAM there's the additional scenario where:
1.  a130xe enables an additional extended RAM bank at 0x4000-0x7fff (where also the space pointer should be readback by ANTIC thru a separate access bit);
2. For a800xl and onward there's eventually the option to install exotic components such as RAMBO XL, which is even more RAM installed at 0x4000-0x7fff and the catch is that it makes sure the internal a130xe RAM is still used i.e. banks $0-$3 internal, $4-$f RAMBO.